rcl/test/rcl/test_events.cpp

```diff
@@ -0,0 +1,553 @@
+    ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Primitives);
+  }
+
+  rcl_ret_t setup_publisher(
+    const rmw_time_t & deadline,
+    const rmw_time_t & lifespan,
+    const rmw_time_t & liveliness_lease_duration,
+    const rmw_qos_liveliness_policy_t liveliness_policy)
+  {
+    // init publisher
+    publisher = rcl_get_zero_initialized_publisher();
+    rcl_publisher_options_t publisher_options = rcl_publisher_get_default_options();
+    publisher_options.qos.deadline = deadline;
+    publisher_options.qos.lifespan = lifespan;
+    publisher_options.qos.liveliness = liveliness_policy;
+    publisher_options.qos.liveliness_lease_duration = liveliness_lease_duration;
+    return rcl_publisher_init(&publisher, this->node_ptr, this->ts, this->topic,
+             &publisher_options);
+  }
+
+  rcl_ret_t setup_subscriber(
+    const rmw_time_t & deadline,
+    const rmw_time_t & lifespan,
+    const rmw_time_t & liveliness_lease_duration,
+    const rmw_qos_liveliness_policy_t liveliness_policy)
+  {
+    // init publisher
+    subscription = rcl_get_zero_initialized_subscription();
+    rcl_subscription_options_t subscription_options = rcl_subscription_get_default_options();
+    subscription_options.qos.deadline = deadline;
+    subscription_options.qos.lifespan = lifespan;
+    subscription_options.qos.liveliness = liveliness_policy;
+    subscription_options.qos.liveliness_lease_duration = liveliness_lease_duration;
+
+    return rcl_subscription_init(&subscription, this->node_ptr, this->ts, this->topic,
+             &subscription_options);
+  }
+
+  void setup_publisher_and_subscriber(
+    const rcl_publisher_event_type_t & pub_event_type,
+    const rcl_subscription_event_type_t & sub_event_type)
+  {
+    rcl_ret_t ret;
+
+    rmw_time_t lifespan {0, 0};
+    rmw_time_t deadline {DEADLINE_PERIOD_IN_S.count(), 0};
+    rmw_time_t lease_duration {LIVELINESS_LEASE_DURATION_IN_S.count(), 0};
+    rmw_qos_liveliness_policy_t liveliness_policy = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC;
+
+    // init publisher
+    ret = setup_publisher(deadline, lifespan, lease_duration, liveliness_policy);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+    // init publisher events
+    publisher_event = rcl_get_zero_initialized_event();
+    ret = rcl_publisher_event_init(&publisher_event, &publisher, pub_event_type);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+    // init subscription
+    ret = setup_subscriber(deadline, lifespan, lease_duration, liveliness_policy);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+    // init subscription event
+    subscription_event = rcl_get_zero_initialized_event();
+    ret = rcl_subscription_event_init(&subscription_event, &subscription, sub_event_type);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+    // wait for discovery
+    // total wait time of 10 seconds, if never ready
```

Comment:
I'm fine with what ever value (I just want it to be consistent with the comment), though making it larger is probably ok, since it will avoid flaky behavior more often and still should run quickly most of the time.

Reply:
I have increased the `max_iterations` to 1000 to match the total wait time of 10 seconds.

Reply:
@mm318 If you think the OSX connext failure is a timeout issue then I recommend upping this wait time (30 seconds?) before re-running